### ISSUE TITLE
Blocks: Refactor away from deprecated component lifecycle methods

### ISSUE THREAD

**Issue by coreymckrill** [WordPress/wordcamp.org#63]
In four of our blocks, we're using some React component lifecycle methods that have been [deprecated](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) in the version of React that Gutenberg is now using. The deprecated methods are:

* **componentWillMount**
* **componentWillReceiveProps**
* componentWillUpdate

We are currently using the ones in **bold**. The deprecation notice linked above has some suggestions for migrating away from the deprecated methods.

Refactoring that may also resolve [the question about `isStillMounted`](https://github.com/WordPress/wordcamp.org/pull/33#discussion_r266171207). If it doesn't, we should open another issue for that.

**Comment by iandunn** [comment]
Updated the description to include https://github.com/WordPress/wordcamp.org/pull/33#discussion_r266171207, since that was resolved in favor of this issue.

**Comment by vedanshujain** [comment]
Reopening because I missed `componentWillReceiveProps` in Sponsors block

**Comment by iandunn** [comment]
Also,

- [ ] Document non-obvious `this.state =` vs `setState()` behavior:  https://github.com/WordPress/wordcamp.org/pull/66#discussion_r272727824

**Comment by coreymckrill** [comment]
This is fully addressed in #69, which also no longer uses `getDerivedStateFromProps`.

**Comment by coreymckrill** [comment]
This was resolved by some of the changes in #69.
